exporter/prometheusremotewriteexporter/helper.go

```diff
@@ -5,30 +5,41 @@ package prometheusremotewriteexporter // import "github.com/open-telemetry/opent
 
 import (
 	"errors"
+	"math"
 	"sort"
 
 	"github.com/prometheus/prometheus/prompb"
 )
 
+// Track batch sizes sent to avoid over allocating huge buffers.
```

Comment:
I added this memoization here to quickly test the theory that this would help with our memory allocations. Let me know if there is a better place to store this state.

Reply:
We want to memoize the sizes between requests so that we can avoid allocating huge buffers beyond the very first export where we will do so since we have no information to decide on the best buffer sizes.

This memoization just stores the estimated best next buffer size and can handle parallel assignments since it is just a single integer assignment. If there is a race we will just use which ever value wins for the next buffer size.